Set/Set/SetGame.swift

```diff
@@ -7,37 +7,53 @@
 
 import Foundation
 
+enum CardState {
+    case chosen, matched, unmatched
+}
+
 final class SetGame {
 
     var canDrawCards: Bool {
-        self.shownCards.count < 24
+        (self.shownCards.compactMap { $0 }.count < 24 ||
+         self.matchedCardsIndices.count > 2) &&
+        self.deck.count > 2
```

Comment:
This may be written in a better way but I don't see it right now

Reply:
piu

Reply:
I think yours does not cover case a here:

> 
When the Deal 3 More Cards button is pressed either a) replace the selected cards if they are a match or b) add 3 cards to the game.